File: practice_7/p2/code/HW_practice-7_p2.cpp
#include <stdio.h>
#include <iostream>
#include <set>
// "Set with the Largest Number of Elements Printer" is S.L.N.E.P.
void SLNRP(std::set<int> setA, std::set<int> setB)
{
    unsigned long len_setA = setA.size();
    unsigned long len_setB = setB.size();
    if (len_setA > len_setB)
    {
        std::cout << "Множество A:\n";
        for (int element : setA) {
            std::cout << element << " ";
        }
        std::cout << std::endl;
    }
    else if (len_setA < len_setB)
    {
        std::cout << "Множество B:\n";
        for (int element : setB) {
            std::cout << element << " ";
        }
        std::cout << std::endl;
    }
    else
    {
        std::cout << "\nКол-во элементов в обоих множествах одинаково.\nВывожу оба множества:\n";
        std::cout << "Множество A:\n";
        for (int element : setA) {
            std::cout << element << " ";
        }
        std::cout << std::endl;
        
        std::cout << "\nМножество B:\n";
        for (int element : setA) {
            std::cout << element << " ";
        }
        std::cout << std::endl;
    }
}

int main()
{
    std::set<int> sA{3 ,6, 7, 8, 9};
    std::set<int> sB{2, 3, 6, 7, 4, 5};
    SLNRP(sA, sB);
    return 0;
}

File: practice_7/p2/code/HW_practice-7_p2.py
# "Set with the Largest Number of Elements Printer" is S.L.N.E.P.
def SLNEP(setA, setB):
    len_setA = len(setA)
    len_setB = len(setB)
    if len_setA > len_setB:
        return print(f'\n\tМножество A:\n  {setA}')
    elif len_setA < len_setB:
        return print(f'\n\tМножество B:\n  {setB}')
    else:
        return print(f'\nКол-во элементов в обоих множествах одинаково.'
                  f'\nВывожу оба множества:'
                  f'\n\tмножество A:\n  {setA} \n\tМножество B:\n  {setB}')

A = {5,4,7,6,8,9}
B = {3,5,6,7,8}
SLNEP(A, B)


